sky/adaptors/kubernetes.py

```diff
@@ -33,8 +33,11 @@
 def _decorate_methods(obj, decorator):
     for attr_name in dir(obj):
         attr = getattr(obj, attr_name)
-        if callable(attr) and not attr_name.startswith('__'):
-            setattr(obj, attr_name, decorator(attr))
+        if callable(attr) and not attr_name.startswith('__') and not getattr(
+                attr, '_is_sky_decorated', False):
+            decorated_attr = decorator(attr)
+            decorated_attr._is_sky_decorated = True  # pylint: disable=protected-access
```

Comment:
It is a bit weird to skip the decoration if it is ever decorated with this method, and `_is_sky_decorated` feels a bit too general. Is it possible to make it `_sky_decorator_types: Set[str]` which stores the purpose for the decoration eg `api_log`.

Also, can we add some comments for why we use `startswith('__')` in the condition?